manuscript/book.md

```diff
@@ -138,32 +138,32 @@ products.map(product => {
 });
 ```
 
-Here’s the `.map()` method is used to _reduce_ an array to a single value by having a side effect instead of returning a new item value from the callback function.
+Here, the `.map()` method is used to _reduce_ an array to a single value by having a side effect instead of returning a new item value from the callback function.
 
-It’s hard to say what this code is doing and it feels like there’s a bug: it only cares about the latest product in a list.
+It’s hard to say what this code is doing, and it feels like there’s a bug: it only cares about the latest product in a list.
 
-If there’s a bug and we need to check if _some_ of the products have expected type, then the `.some()` array method would be the best choice:
+If it’s indeed a bug, and the intention is to check if _some_ of the products have the expected type, then the `.some()` array method would be the best choice:
 
 ```js
 const isExpectedType = products.some(
   product => product.type === expectedType
 );
 ```
 
-If the original code has no bug, then we don’t need to iterate at all and can check the latest array item directly:
+If the behavior of the original code wasn’t a bug, then we actually don’t need to iterate at all. We can check the latest array item directly:
 
 ```js
 const isExpectedType =
   products[products.lengths - 1].type === expectedType;
 ```
 
-Both refactored versions make the intention of the code clearer and leave less doubts that the code is correct. We can probably make the `isExpectedType` variable name more clear, especially in the second refactoring.
+Both refactored versions make the intention of the code clearer and leave fewer doubts that the code is correct. We can probably make the `isExpectedType` variable name more explicit, especially in the second refactoring.
 
 ### Dealing with side effects
 
 Side effects make code harder to understand because you can no longer treat a function as a black box: a function with side effects doesn’t just transform input to output but can affect the environment in unpredictable ways. Functions with side effects are also hard to test because you’ll need to recreate the environment before each test is run and verify it after.
 
-All array methods mentioned in the previous section, except `.forEach()`, imply that they don’t have side effects and that only the return value is used. Introducing any side effects into these methods would make code easy to misread since readers wouldn’t expect to see side effects.
+All array methods mentioned in the previous section, except `.forEach()`, imply that they don’t have side effects and that only the return value is used. Introducing any side effects into these methods would make code easy to misread since readers won’t be expecting them.
```

Comment:
Wouldn't side effects be the object here and the array methods the subject, so them should refer to the object? I wouldn't interpret them as referring to the methods in this sentence, personally, but I'm not a grammar expert either! It was more a case really of trying to avoid the repetition of "side effects".

```suggestion
All array methods mentioned in the previous section, except `.forEach()`, imply that they are pure and only the return value is used. Readers won't be expecting side effects, so introducing them would make code easy to misread.
```

Reply:
I think that's too much to ask of the reader ;-) I usually use "them" only if it's super clear what it is, otherwise expanding it.

Reply:
Let's bring "side effects" back and I'll merge it.

```suggestion
All array methods mentioned in the previous section, except `.forEach()`, imply that they don’t have side effects and that only the return value is used. Introducing any side effects into these methods would make code easy to misread since readers won’t be expecting side effects.
```